### PR TITLE
 Add missing query parameters to members list endpoint

### PR DESCRIPTION
**Pull request by NerdyPepper**
Should take care of #120

This is my first PR to `ruma-client-api`, I hope everything is in order!

### CODE REVIEW

File: src/r0/membership/get_member_events.rs [hunk 1]
```diff
@@ -28,3 +48,23 @@ ruma_api! {
 
     error: crate::Error
 }
+
+/// The kind of membership events to filter for.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum MembershipEventFilter {
```
Comment (jplatte) [inline src/r0/membership/get_member_events.rs]: If you use `#[serde(rename_all = "lowercase")]` on the struct, then you don't need individual `#[serde(rename)]` attributes on the variants.

**Comment by jplatte** [comment]
Please update the link at the top of the file to r0.6.0 (and also drop the `.html`, the canonical form now seems to be the one without it) when updating an endpoint.

Also, as mentioned on #126, please reference other GitHub issue (by number) only in GitHub PR / issue descriptions and comments, not in commit messages. We might change our platform at some point, which would then "invalidate" our commit messages otherwise.

File: src/r0/membership/get_member_events.rs (hunk 1 above)
Comment (NerdyPepper) [reply src/r0/membership/get_member_events.rs]: Done!

**Comment by NerdyPepper** [comment]
> Please update the link at the top of the file to r0.6.0

Oops! I missed that section of `CONTRIBUTING.md`.

> please reference other GitHub issue (by number) only in GitHub [...]

Gotcha! Unfortunately I saw your comment only after adding the commit :/

**Comment by jplatte** [comment]
Thanks, looking good now!

I wonder why the membership state filter doesn't have a `knock` variant like the membership state event. If it did have that, we could just reuse the enum from `ruma_events`. Mind opening an issue on [matrix-doc](https://github.com/matrix-org/matrix-doc/) for that?

**Comment by NerdyPepper** [comment]
@jplatte sure! I have opened https://github.com/matrix-org/matrix-doc/issues/2476.

**Comment by NerdyPepper** [comment]
Quick update from @turt2live:

> Knock is a reserved keyword and has no meaning: https://github.com/matrix-org/matrix-doc/issues/932

**Pull request merged**